src/components/Icon/index.js

```diff
@@ -1,29 +1,38 @@
-import React from 'react';
+import React, { useState } from 'react';
 import PropTypes from 'prop-types';
 
 function Icon({
   className,
   family,
   type,
+  hoverIconType,
   ...rest
 }) {
-  function getCSSClasses() {
+  const [isHovered, setIsHovered] = useState(false);
+  function getCSSClasses(showHoverIcon) {
     const fontFamily = family || 'fa';
 
     return [
       'tyk-icon',
       fontFamily,
-      `${fontFamily}-${type}`,
+      `${fontFamily}-${showHoverIcon ? hoverIconType : type}`,
       className,
     ].filter(Boolean).join(' ');
   }
 
-  return <i className={getCSSClasses()} {...rest} />;
+  const toggleHover = () => setIsHovered(!isHovered);
+
+  return (
+    <span onMouseEnter={toggleHover} onMouseLeave={toggleHover}>
+      {isHovered && hoverIconType ? <i className={getCSSClasses(true)} {...rest} /> : <i className={getCSSClasses()} {...rest} />}
```

Comment:
Having boolean values for parameters is in some cases considered unreadable.
For example `setIsLoading(true)` is considered readable because it is understood that `true` is just the new value.
While something like `getCSSClasses(true)` is not as readable because you can't know what that does without reading the implementation.